### DISCUSSION TITLE
Reserved identifiers

### DISCUSSION THREAD

**Discussion by p0lunin** [tact-lang/spec#5] [Polls]
Reserved identifiers is an identifiers that cannot be used as variable/function names due to they "reserved" e.g. are taken from the pool of the allowed names. At my opinion we must to reserve all popular keywords such as `type`, `class`, `async`, `void` and etc. to save begginers from misunderstanding when read the code. Some professional programmers can use these words for compile-time functions, which can be confused begginers as if these functions are built-in and doing something them expect for but these functions behaviour may not correspond to the common meanings of these words.

**Comment by oleganza** [comment]
I think it's a bit too early to talk about reserved identifiers. When we have a rough draft of a spec - we'll have a visibility into what features/words we have in mind for the future and could reserve those. 

After all, we are going to version-lock each source file for deterministic builds, so legacy code that happens to use some keyword will still compile, while the new one would have to update the code or wrap the conflicting identifiers into backticks or `r#__`. We'll have to support these cases anyway, so i wouldn't worry too much about reserving some words upfront.
